src/modules/Game.class.js

```diff
@@ -1,68 +1,265 @@
 'use strict';
 
-/**
- * This class represents the game.
- * Now it has a basic structure, that is needed for testing.
- * Feel free to add more props and methods if needed.
- */
 class Game {
-  /**
-   * Creates a new game instance.
-   *
-   * @param {number[][]} initialState
-   * The initial state of the board.
-   * @default
-   * [[0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0],
-   *  [0, 0, 0, 0]]
-   *
-   * If passed, the board will be initialized with the provided
-   * initial state.
-   */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  static Status = {
+    idle: 'idle',
+    playing: 'playing',
+    win: 'win',
+    lose: 'lose',
+  };
+
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.state = initialState.map((row) => [...row]);
+    this.status = Game.Status.idle;
+    this.initialState = initialState;
+    this.score = 0;
+  }
+
+  moveLeft() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
+
+      for (let r = 0; r < 4; r++) {
+        const values = [];
+
+        for (let c = 0; c < 4; c++) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
```

Comment:
There is a potential bug here. After merging two cells, you should skip the next cell to prevent it from merging with another cell in the same move. Otherwise, you might merge more than once in a single move which is not allowed.